enhancements/kube-apiserver/audit-policy.md

```diff
@@ -0,0 +1,428 @@
+  i.e. it allows them to increase audit verbosity
+- without risking to log security-sensitive resources
+- and it is feasible to maintain the feature in the future when the dynamic audit API takes over.
+
+With performance tests we have proven that there is no strong reason to pre-filter audit logs before
+writing them to disk. Filtering can be done by custom solutions implemented by customers, or by
+using external log processing systems. Hence, we strictly separate filtering from setting the audit log 
+depth using the new API.
+
+### Goals
+
+1. add an abstract audit policy configuration to `apiservers.config.openshift.io/v1` with few predefined policies
+  - starting with different audit depths policies only,
+  - but stay extensible in the future if necessary.
+2. applied by kube-apiserver, openshift-apiserver (and soon oauth-apiserver).
+
+### Non-Goals
+
+- configuration of audit backends like webhooks
+- configuration of file logger parameters
+- configuration of filtering (like by resource, API group, namespaces, etc.)
+- configuration of what we log in each event (other than the depth).
+
+## Proposal
+
+We propose to add an `audit.profile` field to `apiservers.config.openshift.io/v1` which selects the 
+audit policy to be deployed to all OpenShift-provided API servers in the cluster. From the beginning
+we provide the following profiles:
+
+- `Default`: this is the existing default policy. `Default` is also the default value in the `APIServer` custom resource:
+
+  ```yaml
+    apiVersion: audit.k8s.io/v1beta1
+    kind: Policy
+    rules:
+    # Don't log requests for events
+    - level: None
+      resources:
+      - group: ""
+        resources: ["events"]
+    # Don't log oauth tokens as metadata.name is the secret
+    - level: None
+      resources:
+      - group: "oauth.openshift.io"
+        resources: ["oauthaccesstokens", "oauthauthorizetokens"]
+    # Don't log authenticated requests to certain non-resource URL paths.
+    - level: None
+      userGroups: ["system:authenticated", "system:unauthenticated"]
+      nonResourceURLs:
+      - "/api*" # Wildcard matching.
+      - "/version"
+      - "/healthz"
+      - "/readyz"
+    # Log the full Identity API resource object so that the audit trail
+    # allows us to match the username with the IDP identity.
+    - level: RequestResponse
+      verbs: ["create", "update", "patch"]
+      resources:
+      - group: "user.openshift.io"
+        resources: ["identities"]
+  
+    # Here the tail begins which differs per profile:
+  
+    # A catch-all rule to log all other requests at the Metadata level.
+    - level: Metadata
+      # Long-running requests like watches that fall under this rule will not
+      # generate an audit event in RequestReceived.
+      omitStages:
+      - RequestReceived
+   ```
+
+- `WriteRequestBodies`: this is like `Default`, but it logs request and response HTTP payloads for write requests (create, update, patch).
```

Comment:
`MutatingRequestBodies`

Reply:
We settled on `WriteRequestBodies` in Slack.